Jenkinsfile

```diff
@@ -0,0 +1,69 @@
+#!groovy
+
+def python_versions = ['2.7','3.3','3.4','3.5','3.6','pypy','latest']
+def frameworks = ['django-1.8', 'django-1.9', 'django-1.10', 'django-1.11', 'django-master']
+def exclude_jobs = ['2.7_django-master', '3.3_django-1.9', '3.3_django-1.10', '3.3_django-1.11', '3.3_django-master']
+def test_jobs = [:]
+def linters = ['isort', 'flake8', 'docs']
+def linter_jobs = [:]
+
+node{
+    withEnv(["HOME=/var/lib/jenkins",
+             "PIP_CACHE=.cache/pip"
+             ]) {
+        stage('Checkout'){
+            node('linux'){
+                sh 'echo "do some SCM checkout..."'
+                checkout scm
+            }
+        }
+       
+        stage('Lint'){
+            def label="build_no=${env.BUILD_ID}"
+            def pip_cache = "${env.WORKSPACE}/${env.PIP_CACHE}"
+            linters.each {
+                linter_jobs["${it}"] = {
+                    node('linux'){
+                        try{
+                            checkout scm
+                            sh("./tests/scripts/docker/${it}.sh ${label} ${pip_cache}")
+                        }catch(e){
+                            currentBuild.result = 'FAILURE'
+                            throw e
+                        }finally{
+                            sh("./tests/scripts/docker/remove.sh ${label}")
+                        }
+                    }
+                }
+            }
+            parallel linter_jobs
+        }
+        
+        stage("Test Run"){
+            if (currentBuild.result.equals('FAILURE')){
+                return
+            }
+            python_versions.each{ py_ver -> 
+                frameworks.each{ framework -> 
+                    def job = "${py_ver}_${framework}".toString()
+                    if(exclude_jobs.contains(job)){
+                        return
+                    }
+                    test_jobs[job] = {
+                        node('linux'){
```

Comment:
I assume at a later stage here also `node('windows')` or similar could be added?

Reply:
That is the plan yes, but as you said on a later stage, since we also have to figure out how to handle it with dependencies/docker. 
Right now the goal is to have the same testsuite as we have on travis, and also enable usage of docker for local testing.